src/sparseml/pytorch/sparsification/pruning/modifier_pruning_base.py

```diff
@@ -117,10 +117,10 @@ def __init__(
         log_types: Union[str, List[str]] = None,
         global_sparsity: bool = False,
         allow_reintroduction: bool = False,
+        leave_enabled: bool = False,
```

Comment:
noting that these changes will be replaced by multi-inheretance patch discussed earlier today

Reply:
@dbogunowicz bumping this one - is this still necessary / does anything break by adding this now?

Reply:
`leave_enabled` works correctly, as expected, and is compatible with Konstantin's changes as well as all the tests :)

Reply:
...and also necessary imo:
- setting `leave_enabled=False` functionality is helpful to reproduce the original paper, where after alternating between compression/decompression phases with `update_frequency=5`, we finish off the training with `10 decompression` and finally `15 compression` fine-tuning epochs. So in that case, it is useful to stop masking the weights after last epoch of the modifier, just before the 10-epoch fine-tuning decompression phase.
- setting `leave_enabled=True` would be desirable by most of the users to simply use this modifier out-of-the-box and retrieve the sparse model.

Also, in the paper Dan's students experiment with having dense-only/sparse-only fine tuning after the compression/decompression phases, so all in all, I like having this argument there.

Having said that, if you agree with my arguments, I should probably make this argument `True` by default (to make it easier to use in out-of-the-box fashion).